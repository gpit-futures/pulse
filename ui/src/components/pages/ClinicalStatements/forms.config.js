export const valuesNames = {
  TYPE: 'type',
  STATEMENT: 'text',
  NOTE: 'clinicalNote',
  NOTE_CONTENT: 'contentStore',
  NOTE_TEXT: 'text',
  EDITABLE_CLASS: '.editable',
  EDITABLE_EMPTY_CLASS: '.editable-empty',
  EDITABLE_EMPTY_FIELDS: 'emptyFieldsAmount',
  TAGS: 'tags',
  DATE_CREATED: 'dateCreated',
  AUTHOR: 'author',

  SOURCE: 'source',
  SOURCE_ID: 'sourceId',
};

export const valuesLabels = {
  TYPE: 'Type',
  STATEMENT: 'Statement',
  STATEMENTS: 'Statements',
  NOTE: 'Clinical Note',
  TAGS: 'Tags',
  AUTHOR: 'Author',
  DATE: 'Date',
  SEARCH: 'Search',

  SOURCE: 'Source',
  SOURCE_ID: 'Source ID',
};
