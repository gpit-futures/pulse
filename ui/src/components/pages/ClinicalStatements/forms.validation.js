import { valuesNames } from './forms.config';

const validateForm = (values) => {
  const errors = {};
  errors[valuesNames.TYPE] = !values[valuesNames.TYPE] ? 'You must enter a value.' : null;

  if (!values[valuesNames.NOTE] || !values[valuesNames.NOTE][valuesNames.NOTE_TEXT]) {
    errors[valuesNames.NOTE] = 'You must enter a value.';
  } else if (values[valuesNames.NOTE][valuesNames.EDITABLE_EMPTY_FIELDS] !== 0) {
    errors[valuesNames.NOTE] = 'You must enter a value to a statement instead of \'?\' symbol or \'Empty\' value.';
  } else {
   errors[valuesNames.NOTE] = null;
  }

  return errors;
};

export { validateForm }
