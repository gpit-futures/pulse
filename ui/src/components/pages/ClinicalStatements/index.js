import { combineEpics } from 'redux-observable';

import asyncComponent from '../../../components/containers/AsyncComponent/AsyncComponent';
import { clientUrls } from '../../../config/client-urls.constants';

import { fetchPatientClinicalStatementsDetailEpic } from './ducks/fetch-patient-clinical-statements-detail.duck';
import { fetchPatientClinicalStatementsEpic } from './ducks/fetch-patient-clinical-statements.duck';
import { fetchPatientClinicalStatementsTagsEpic } from './ducks/fetch-patient-clinical-statements-tags.duck';
import { fetchPatientClinicalStatementsQueryEpic } from './ducks/fetch-patient-clinical-statements-query.duck';
import { fetchPatientClinicalStatementsCreateEpic } from './ducks/fetch-patient-clinical-statements-create.duck';

import patientsClinicalStatements from './ducks/fetch-patient-clinical-statements.duck';
import patientsClinicalStatementsTags from './ducks/fetch-patient-clinical-statements-tags.duck';
import patientsClinicalStatementsQuery from './ducks/fetch-patient-clinical-statements-query.duck';
import patientClinicalStatementsCreate from './ducks/fetch-patient-clinical-statements-create.duck';
import clinicalStatementsDetail from './ducks/fetch-patient-clinical-statements-detail.duck';

const epics = combineEpics(
  fetchPatientClinicalStatementsDetailEpic,
  fetchPatientClinicalStatementsEpic,
  fetchPatientClinicalStatementsTagsEpic,
	fetchPatientClinicalStatementsQueryEpic,
  fetchPatientClinicalStatementsCreateEpic
);

const ClinicalStatements = asyncComponent(() => import(/* webpackChunkName: "clinicalStatements" */ './ClinicalStatements').then(module => module.default));

const reducers = {
  patientsClinicalStatements,
  patientClinicalStatementsCreate,
  clinicalStatementsDetail,
	patientsClinicalStatementsTags,
	patientsClinicalStatementsQuery,
};

const sidebarConfig = { key: 'clinicalStatements', pathToTransition: '/clinicalStatements', name: 'Clinical Statements', isVisible: true };

const routers = [
  { key: 'clinicalStatements', component: ClinicalStatements, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.CLINICAL_STATEMENTS}` },
  { key: 'clinicalStatementsCreate', component: ClinicalStatements, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.CLINICAL_STATEMENTS}/create` },
  { key: 'clinicalStatementsDetail', component: ClinicalStatements, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.CLINICAL_STATEMENTS}/:sourceId` },
];

export default {
  component: ClinicalStatements,
  epics, reducers, sidebarConfig, routers,
}

