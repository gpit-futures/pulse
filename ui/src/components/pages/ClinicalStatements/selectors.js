import { createSelector } from 'reselect';
import _ from 'lodash/fp';

const clinicalStatementsCreateFormSelector = _.getOr({}, 'form.clinicalStatementsCreateFormSelector');

const patientClinicalStatementsSelector = createSelector(
  ({ patientsClinicalStatements }) => patientsClinicalStatements,
  (state, props) => _.getOr(null, 'match.params.userId', props),
  (patientsClinicalStatements, userId) => {
    const allClinicalStatements = patientsClinicalStatements[userId];
    return ({ allClinicalStatements, userId });
  }
);

const clinicalStatementsCreateFormStateSelector = createSelector(clinicalStatementsCreateFormSelector,
  clinicalStatementsCreateFormState => ({ clinicalStatementsCreateFormState }));

const patientClinicalStatementsDetailSelector = createSelector(
  ({ clinicalStatementsDetail }) => clinicalStatementsDetail,
  (state, props) => _.getOr(null, 'match.params.userId', props),
  (clinicalStatementsDetail, userId) => {
    const clinicalStatementDetail = clinicalStatementsDetail[userId];
    return ({ clinicalStatementDetail, userId });
  }
);

const patientClinicalStatementsTagsSelector = createSelector(
	({ patientsClinicalStatementsTags }) => patientsClinicalStatementsTags,
	(state, props) => _.getOr(null, 'match.params.userId', props),
	(patientsClinicalStatementsTags, userId) => {
		const clinicalStatementsTags = patientsClinicalStatementsTags[userId];
		return ({ clinicalStatementsTags, userId });
	}
);

const patientClinicalStatementsQuerySelector = createSelector(
	({ patientsClinicalStatementsQuery }) => patientsClinicalStatementsQuery,
	(state, props) => _.getOr(null, 'match.params.userId', props),
	(patientsClinicalStatementsQuery, userId) => {
		const clinicalStatementsQuery = patientsClinicalStatementsQuery;
		return ({ clinicalStatementsQuery, userId });
	}
);

export {
	patientClinicalStatementsSelector,
	clinicalStatementsCreateFormStateSelector,
	patientClinicalStatementsDetailSelector,
	patientClinicalStatementsTagsSelector,
	patientClinicalStatementsQuerySelector
}
