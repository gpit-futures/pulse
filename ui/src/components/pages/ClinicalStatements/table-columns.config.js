import { getDDMMMYYYY } from '../../../utils/time-helpers.utils';


export const columnsConfig = [
  { key: 'type', title: 'Type', width: '30%' },
  { key: 'dateCreated', title: 'Date Created', transformer: getDDMMMYYYY, width: '30%' },
  { key: 'author', title: 'Author', width: '20%' },
  { key: 'source', title: 'Source', width: '20%' },
  { key: 'sourceId', title: 'SourceID', display: 'none' },
];


export const defaultColumnsSelected = {
  type: true,
  dateCreated: true,
  author: true,
  source: true,
  sourceId: true,
};
