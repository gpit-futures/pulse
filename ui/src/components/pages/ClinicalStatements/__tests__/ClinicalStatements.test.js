import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import ClinicalStatements from '../ClinicalStatements';
import { valuesNames } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

// frequently used variables
const userId = '9999999000';
const sourceId = 'acb0eaf2-d1df-4c7a-9382-619b31935f2b';

// storage setup in different configurations
const mockStore = configureStore();
const storeResource = {
  patientsClinicalStatements: {
    '9999999024': [
      {
        [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
        [valuesNames.DATE_CREATED]: 1510224834000,
        [valuesNames.TYPE]: '11',
        [valuesNames.SOURCE]: 'ethercis',
        [valuesNames.SOURCE_ID]: 'd6b4d397-779c-4a4e-b5eb-2084fb11876f',
        dateCreatedConvert: '09-Nov-2017',
      },
      {
        [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
        [valuesNames.DATE_CREATED]: 1510224834000,
        [valuesNames.TYPE]: '11',
        [valuesNames.SOURCE]: 'ethercis',
        [valuesNames.SOURCE_ID]: 'd6b4d397-779c-4a4e-b5eb-2084fb11876f',
        dateCreatedConvert: '09-Nov-2017',
      },
    ],
  },
};
const storeWithFormsError = mockStore(Object.assign({
  clinicalStatementsDetail: {
    '9999999000': {
      [valuesNames.TAGS]: { chestpain: true, shouder: true },
      [valuesNames.STATEMENT]: '<span class="tag" data-tag-id="1519236197404" data-id="7" data-phrase="The pain was mild at |?| in severity" contenteditable="false">The pain was mild at <span class="editable editable-click editable-unsaved editable-empty editable-open" contenteditable="false" data-arr-subject="The pain was mild at " editable-text="" data-arr-unit="?" data-arr-value=" in severity" data-original-title="" title="" style="background-color: rgba(0, 0, 0, 0);" aria-describedby="popover368681">Empty</span><div class="popover editable-container editable-popup fade top in" role="tooltip" id="popover368681" style="top: -74px; left: 12.6719px; display: block;"><div class="arrow" style="left: 50%;"></div><h3 class="popover-title">Edit Text</h3><div class="popover-content"> <div><div class="editableform-loading" style="display: none;"></div><form class="form-inline editableform" style=""><div class="control-group form-group"><div><div class="editable-input" style="position: relative;"><input type="text" class="form-control input-sm" style="padding-right: 24px;"><span class="editable-clear-x" style=""></span></div><div class="editable-buttons"><button type="submit" class="btn btn-primary btn-sm editable-submit"><i class="glyphicon glyphicon-ok"></i></button><button type="button" class="btn btn-default btn-sm editable-cancel"><i class="glyphicon glyphicon-remove"></i></button></div></div><div class="editable-error-block help-block" style="display: none;"></div></div></form></div></div></div> in severity. <a class="remove" contenteditable="false"><i class="fa fa-close" contenteditable="false"></i></a></span> <span id="temp" contenteditable="false"></span>',
      [valuesNames.TYPE]: 'Test name',
      [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
      [valuesNames.DATE_CREATED]: 1519236225009,
      [valuesNames.SOURCE]: 'qewdDB',
      [valuesNames.SOURCE_ID]: 'd9acb27f-6688-4515-ab0a-5db636f36f05',
    },
  },
  form: {
    clinicalStatementsCreateFormSelector: {
      syncErrors: {
        [valuesNames.TYPE]: 'You must enter a value.',
        [valuesNames.NOTE]: 'You must enter a value.',
      },
    },
  },
}, storeResource));
const storeEmpty = mockStore(Object.assign({
  clinicalStatementsDetail: {
    '9999999000': {},
  },
}, storeResource));
const storeWithDetail = mockStore(Object.assign({
  clinicalStatementsDetail: {
    '9999999000': {
      [valuesNames.TAGS]: { chestpain: true },
      [valuesNames.STATEMENT]: 'The pain was mild at 3 in severity',
      [valuesNames.TYPE]: 'Test name',
      [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
      [valuesNames.DATE_CREATED]: 1519236225009,
      [valuesNames.SOURCE]: 'qewdDB',
      [valuesNames.SOURCE_ID]: 'd9acb27f-6688-4515-ab0a-5db636f36f05',
    },
  },
}, storeResource));

// configure context for various tests
const generateNewContext = (oldContext, pathname) => {
  const newContext = Object.assign({}, oldContext);
  newContext.router = Object.assign({}, newContext.router);
  newContext.router.history = Object.assign({}, newContext.router.history);
  newContext.router.history.location = { pathname };
  return newContext;
};
const context = {
  router: {
    history: {
      push: () => {},
      replace: () => {},
      location: {
        pathname: `/patients/${userId}/clinicalStatements`,
      },
    },
    route: {
      match: {
        params: {
          sourceId,
          userId,
        },
      },
    },
  },
};
const contextCreate = generateNewContext(context, `/patients/${userId}/clinicalStatements/create`);
const contextDetail = generateNewContext(context, `/patients/${userId}/clinicalStatements/${sourceId}`);

// configuration of forms for testing methods
const formValuesEdit = {
  [valuesNames.TYPE]: '2',
  [valuesNames.NOTE]: 2,
  [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
};
const formValuesCreate = {
  [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
  [valuesNames.SOURCE]: 'openehr',
  [valuesNames.TYPE]: 'test',
  [valuesNames.NOTE]: 'test'
};
const match = {
  params: {
    userId,
  },
};

describe('Component <ClinicalStatements />', () => {
  it('should renders correctly with clinicalStatementPanel and testing Detail Panel', () => {
    const component = shallow(
      <ClinicalStatements
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive();

    // Testing component handleDetailClinicalStatementsClick methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('ClinicalStatementsDetail')).toHaveLength(0);
    expect(component.find('PluginCreate')).toHaveLength(0);

    component.instance().handleDetailClinicalStatementsClick('065d85e3-3cd5-4604-bb94-5685fffb193d');
    const componentStateAfterMethod = component.state();
    component.setState({ isSecondPanel: true, isDetailPanelVisible: true, isBtnExpandVisible: true, isBtnCreateVisible: true, isCreatePanelVisible: false, openedPanel: 'clinicalStatementPanel', editedPanel: {}, expandedPanel: 'all', isLoading: true });
    const componentStateAfterSetState = component.state();

    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('ClinicalStatementsDetail')).toHaveLength(1);
    expect(component.find('PluginCreate')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    // Testing component handleExpand methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.state().openedPanel).toEqual('clinicalStatementPanel');
    expect(component.state().expandedPanel).toEqual('all');
    component.instance().handleExpand('clinicalStatementPanel', 'clinicalStatementsDetail');
    expect(component.state().openedPanel).toEqual('clinicalStatementPanel');
    expect(component.state().expandedPanel).toEqual('clinicalStatementPanel');
    component.setState({ openedPanel: 'clinicalStatementPanel', expandedPanel: 'clinicalStatementPanel' });
    expect(component.find('PluginListHeader')).toHaveLength(0);
    expect(component.find('PluginMainPanel')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    component.setState({ openedPanel: 'clinicalStatementPanel', expandedPanel: 'all' });

    component.instance().handleExpand('clinicalStatementPanel', 'clinicalStatementsCreate');
    component.setState({ expandedPanel: 'all' });
    component.instance().handleExpand('clinicalStatementPanel', 'clinicalStatementsMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('clinicalStatementPanel', 'clinicalStatementsMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('clinicalStatementPanel', 'clinicalStatementsCreate');

    // Testing component detail form mock methods
    component.instance().handleEdit();
    component.instance().handleClinicalStatementDetailCancel();
    component.instance().handleSaveSettingsDetailForm();

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with clinicalStatementsDetail and testing Create Panel', () => {
    const component = shallow(
      <ClinicalStatements
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive();

    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('ClinicalStatementsDetail')).toHaveLength(0);
    expect(component.find('PluginCreate')).toHaveLength(0);

    // Testing component create panel methods
    component.instance().handleCreate();
    const componentStateAfterMethod = component.state();
    component.setState({ isBtnCreateVisible: false, isCreatePanelVisible: true, openedPanel: 'clinicalStatementsCreate', isSecondPanel: true, isDetailPanelVisible: false, isSubmit: false, isLoading: true });
    const componentStateAfterSetState = component.state();
    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('PluginCreate')).toHaveLength(1);
    expect(component).toMatchSnapshot();
    component.instance().handleCreateCancel();
    component.instance().handleSaveSettingsCreateForm(formValuesCreate);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing another methods', () => {
    const component = shallow(
      <ClinicalStatements
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive();

    // Testing component hideCreateForm methods
    component.instance().hideCreateForm();
    expect(component.state().isBtnCreateVisible).toEqual(true);
    expect(component.state().isCreatePanelVisible).toEqual(false);
    expect(component.state().openedPanel).toEqual('clinicalStatementPanel');
    expect(component.state().isSecondPanel).toEqual(false);
    expect(component.state().expandedPanel).toEqual('all');
    expect(component.state().isBtnExpandVisible).toEqual(false);

    // Testing component handleFilterChange methods
    expect(component.state().nameShouldInclude).toEqual('');
    component.instance().handleFilterChange({ target: { value: 'test' } });
    expect(component.state().nameShouldInclude).toEqual('test');
    component.instance().handleFilterChange({ target: { value: '' } });
    expect(component.state().nameShouldInclude).toEqual('');

    // Testing component handleHeaderCellClick methods
    expect(component.state().columnNameSortBy).toEqual('type');
    expect(component.state().sortingOrder).toEqual('asc');
    component.instance().handleHeaderCellClick({}, { name: 'author', sortingOrder: 'desc' });
    expect(component.state().columnNameSortBy).toEqual('author');
    expect(component.state().sortingOrder).toEqual('desc');

    // Testing component handleSetOffset methods
    expect(component.state().offset).toEqual(0);
    component.instance().handleSetOffset(10);
    expect(component.state().offset).toEqual(10);

    expect(component.state().openedPanel).toEqual('clinicalStatementPanel');
    component.instance().handleShow('anotherName');
    expect(component.state().openedPanel).toEqual('anotherName');

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing context and lifecycle componentWillReceiveProps', () => {
    const component = shallow(
      <ClinicalStatements
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive();

    component.setProps({ test: 'testing context' });
    component.setContext(contextCreate);
    component.setProps({ test: 'testing create context' });
    component.setContext(contextDetail);
    component.setProps({ test: 'testing edit context' });
  });

  it('should renders correctly with forms error', () => {
    const component = shallow(
      <ClinicalStatements
        store={storeWithFormsError}
        match={match}
      />, { context }).dive().dive().dive().dive().dive();

    component.instance().handleSaveSettingsCreateForm(formValuesCreate);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with empty store', () => {
    const component = shallow(
      <ClinicalStatements
        store={storeEmpty}
        match={match}
      />, { context }).dive().dive().dive().dive().dive();

    expect(component).toMatchSnapshot();
  });
});
