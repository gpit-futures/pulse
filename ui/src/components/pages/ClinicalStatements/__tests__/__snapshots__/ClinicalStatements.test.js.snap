// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <ClinicalStatements /> should renders correctly and testing another methods 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="clinicalStatementsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="clinicalStatementsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Clinical Statements"
          />
          <PluginMainPanel
            columnNameSortBy="type"
            emptyDataMessage="No clinical statements"
            headers={
              Array [
                Object {
                  "key": "type",
                  "title": "Type",
                  "width": "30%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "30%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "20%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="clinicalStatements"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ClinicalStatements /> should renders correctly with clinicalStatementPanel and testing Detail Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="clinicalStatementsMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="clinicalStatementsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Clinical Statements"
          />
          <PluginMainPanel
            columnNameSortBy="type"
            emptyDataMessage="No clinical statements"
            headers={
              Array [
                Object {
                  "key": "type",
                  "title": "Type",
                  "width": "30%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "30%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "20%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            id="d9acb27f-6688-4515-ab0a-5db636f36f05"
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="clinicalStatements"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <ClinicalStatementsDetail
          currentPanel="clinicalStatementsDetail"
          detail={
            Object {
              "author": "bob.smith@gmail.com",
              "dateCreated": 1519236225009,
              "source": "qewdDB",
              "sourceId": "d9acb27f-6688-4515-ab0a-5db636f36f05",
              "tags": Object {
                "chestpain": true,
              },
              "text": "The pain was mild at 3 in severity",
              "type": "Test name",
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          isSubmit={false}
          name="clinicalStatementsDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="clinicalStatementPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ClinicalStatements /> should renders correctly with clinicalStatementPanel and testing Detail Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section full-panel full-panel-details"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <ClinicalStatementsDetail
          currentPanel="clinicalStatementsDetail"
          detail={
            Object {
              "author": "bob.smith@gmail.com",
              "dateCreated": 1519236225009,
              "source": "qewdDB",
              "sourceId": "d9acb27f-6688-4515-ab0a-5db636f36f05",
              "tags": Object {
                "chestpain": true,
              },
              "text": "The pain was mild at 3 in severity",
              "type": "Test name",
            }
          }
          editedPanel={Object {}}
          expandedPanel="clinicalStatementPanel"
          isSubmit={false}
          name="clinicalStatementsDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="clinicalStatementPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ClinicalStatements /> should renders correctly with clinicalStatementPanel and testing Detail Panel 3`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    />
  </div>
</section>
`;

exports[`Component <ClinicalStatements /> should renders correctly with clinicalStatementsDetail and testing Create Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="clinicalStatementsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="clinicalStatementsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Clinical Statements"
          />
          <PluginMainPanel
            columnNameSortBy="type"
            emptyDataMessage="No clinical statements"
            headers={
              Array [
                Object {
                  "key": "type",
                  "title": "Type",
                  "width": "30%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "30%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "20%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="clinicalStatements"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <PluginCreate
          componentForm={
            <ReduxForm
              clickOnCreate={false}
              isSubmit={false}
              match={
                Object {
                  "params": Object {
                    "userId": "9999999000",
                  },
                }
              }
            />
          }
          currentPanel="clinicalStatementsCreate"
          expandedPanel="all"
          isCreatePanelVisible={true}
          isImport={false}
          name="clinicalStatementsCreate"
          onCancel={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="clinicalStatementsCreate"
          title="Create Clinical Statement"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ClinicalStatements /> should renders correctly with clinicalStatementsDetail and testing Create Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="clinicalStatementsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="clinicalStatementsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Clinical Statements"
          />
          <PluginMainPanel
            columnNameSortBy="type"
            emptyDataMessage="No clinical statements"
            headers={
              Array [
                Object {
                  "key": "type",
                  "title": "Type",
                  "width": "30%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "30%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "20%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="clinicalStatements"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <PluginCreate
          componentForm={
            <ReduxForm
              clickOnCreate={false}
              isSubmit={false}
              match={
                Object {
                  "params": Object {
                    "userId": "9999999000",
                  },
                }
              }
            />
          }
          currentPanel="clinicalStatementsCreate"
          expandedPanel="all"
          isCreatePanelVisible={true}
          isImport={false}
          name="clinicalStatementsCreate"
          onCancel={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="clinicalStatementsCreate"
          title="Create Clinical Statement"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ClinicalStatements /> should renders correctly with empty store 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="clinicalStatementsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="clinicalStatementsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Clinical Statements"
          />
          <PluginMainPanel
            columnNameSortBy="type"
            emptyDataMessage="No clinical statements"
            headers={
              Array [
                Object {
                  "key": "type",
                  "title": "Type",
                  "width": "30%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "30%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "20%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="clinicalStatements"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ClinicalStatements /> should renders correctly with forms error 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="clinicalStatementsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="clinicalStatementsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Clinical Statements"
          />
          <PluginMainPanel
            columnNameSortBy="type"
            emptyDataMessage="No clinical statements"
            headers={
              Array [
                Object {
                  "key": "type",
                  "title": "Type",
                  "width": "30%",
                },
                Object {
                  "key": "dateCreated",
                  "title": "Date Created",
                  "transformer": [Function],
                  "width": "30%",
                },
                Object {
                  "key": "author",
                  "title": "Author",
                  "width": "20%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="clinicalStatements"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;
